exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/dml/MergeBaseTest.kt

```diff
@@ -0,0 +1,82 @@
+package org.jetbrains.exposed.sql.tests.shared.dml
+
+import kotlinx.datetime.LocalDateTime
+import org.jetbrains.exposed.dao.id.IntIdTable
+import org.jetbrains.exposed.sql.*
+import org.jetbrains.exposed.sql.kotlin.datetime.datetime
+import org.jetbrains.exposed.sql.tests.DatabaseTestsBase
+import org.jetbrains.exposed.sql.tests.TestDB
+
+val TEST_DEFAULT_DATE_TIME = LocalDateTime(2000, 1, 1, 0, 0, 0, 0)
+
+abstract class MergeBaseTest : DatabaseTestsBase() {
+    protected fun allDbExcept(includeSettings: List<TestDB>) = TestDB.entries - includeSettings.toSet()
+
+    protected val defaultExcludeSettings = listOf(TestDB.SQLITE, TestDB.MYSQL, TestDB.MARIADB)
+
+    protected fun withMergeTestTables(excludeSettings: List<TestDB> = emptyList(), statement: Transaction.(TestDB) -> Unit) = withTables(
+        defaultExcludeSettings + excludeSettings, Source, Dest
+    ) {
+        statement(it)
+    }
+
+    protected fun withMergeTestTablesAndDefaultData(excludeSettings: List<TestDB> = emptyList(), statement: Transaction.(TestDB) -> Unit) {
+        withMergeTestTables(excludeSettings) {
```

Comment:
I vote for passing `Dest` and `Source` as arguments to the `statement` block so that they can be named and it becomes obvious they are a part of the created test table group and not some external table that is created elsewhere, like how it's done in `withCitiesAndUsers()` and the other test functions:
```kt
withMergeTestTablesAndDefaultData { dest, _ ->
    dest.mergeFrom()
}
```